main.js

```diff
@@ -163,14 +164,21 @@ module.exports.shuttingDown = shuttingDown;
 
 const commandFiles = fs.readdirSync('./commands').filter(file => file.endsWith('.js'));
 for (const file of commandFiles) {
-	/** @type {{data: import('@discordjs/builders').SlashCommandBuilder}} */
+	/** @type {{data: import('@discordjs/builders').SlashCommandBuilder, checks: string[], permissions: {user: string[], bot: string[]}, execute(interaction: import('discord.js').CommandInteraction): Promise<void>}} */
 	const command = require(`./commands/${file}`);
 	bot.commands.set(command.data.name, command);
 }
 
+const buttonFiles = fs.readdirSync('./components/buttons').filter(file => file.endsWith('.js'));
+for (const file of buttonFiles) {
+	/** @type {{name: string, execute(interaction: import('discord.js').ButtonInteraction): Promise<void)>}} */
+	const button = require(`./components/buttons/${file}`);
+	bot.buttons.set(button.name, button);
+}
+
```

Comment:
you should have a componentFiles, and then use a for of loop to iterate through that. in the event we add more things to components in the future, this'll help us a lot. (the dir name that you're iterating through in components can be used to dynamically set it, like for (const file of componentFiles) and then bot[file].set or something like that)